mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -141,6 +135,48 @@ def __init__(self, name=None, bus=None, use_settings=True):
         self.bind(bus)
         #: Mycroft global configuration. (dict)
         self.config_core = Configuration.get()
+
+        self.settings = None
+
+        #: To not break existing setups,
+        #: save to skill directory if the file exists already
+        self.settings_write_path = Path(self.root_dir)
+
+        #: Otherwise save to XDG_CONFIG_DIR
+        if not self.settings_write_path.joinpath('settings.json').exists():
+            self.settings_write_path = Path(BaseDirectory.save_config_path(
+                'mycroft', 'skills', self.name))
+
+        self.settings_read_path = None
+
+        #: To note break existing setups,
+        #: read from skill directory if the settings file exists there
+        self.settings_read_path = Path(self.root_dir)
+
+        #: Then, check XDG_CONFIG_DIR
+        if not self.settings_read_path.joinpath('settings.json').exists():
+            for dir in BaseDirectory.load_config_paths('mycroft',
+                                                       'skills',
+                                                       self.name):
+                path = Path(dir)
+                #: If there is a settings file here, use it
+                if path.joinpath('settings.json').exists():
+                    self.settings_read_path = path
+                    break
+
+        #: Lastly, check /etc/mycroft
+        if not self.settings_read_path.joinpath('settings.json').exists():
+            self.settings_read_path = Path(
+                '/etc/mycroft/skills/').joinpath(self.name)
```

Comment:
Is the /etc/mycroft addition needed?

Reply:
Not necessarily, but that is what most programs on Linux adhere too. Just check your average `/etc` folder on any Linux distributions, most programs have their system-wide settings there.

Reply:
The settings are something the skills handle them selves, read only defaults should really not be introduced imho. Basically the only places I'd expect to see the skill settings would be `$XDG_CONFIG_HOME/...` and the settings dir, none of the read only places is really of interest.

Reply:
Does the settings_read_path need to be a member though? it's read at startup and then the only thing of interest to the skill is the write path

Reply:
Ok, I removed the check for `/etc/mycroft`. Also made `settings_read_path` a local variable only.